### PR TITLE
[Web-15] define layout

### PR DESCRIPTION
**Pull request by emilyPhee**
https://user-images.githubusercontent.com/57051508/211225704-2cbd2f25-9b28-418f-a085-ca8a155fa9fb.mov

- Define top and bottom section
  - placed AddYearDropdown component at top left
- Generate yearOptions array based on current year
  - display 17-18 to 10 years ahead from the current year

### CODE REVIEW

**Comment by emilyPhee** [comment]
[[Web] Position AddYearDropdown component and implement functionalities](https://trello.com/c/zYrKizQv/15-web-position-addyeardropdown-component-and-implement-functionalities)

File: new-nextjs-client/pages/index.module.scss [hunk 1]
```diff
@@ -1,6 +1,19 @@
 .container {
   padding: 0 2rem;
   background: $main-background;
+  display: grid;
+  grid-template-rows: 7rem auto;
+
+  .topLayout {
+    display: flex;
+    justify-content: space-between;
+    .addYear {
```
Comment (anthopark) [inline new-nextjs-client/pages/index.module.scss]: Let's use `.addYearWrapper`. 

Let's append  "Wrapper" to the className of an element that wraps a component and is used for positioning such as padding/margin

File: new-nextjs-client/pages/index.module.scss [hunk 2]
```diff
@@ -1,6 +1,19 @@
 .container {
   padding: 0 2rem;
   background: $main-background;
+  display: grid;
+  grid-template-rows: 7rem auto;
+
+  .topLayout {
```
Comment (anthopark) [inline new-nextjs-client/pages/index.module.scss]: .topSection may be more appropriate

File: new-nextjs-client/pages/index.module.scss [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+    display: flex;
+    justify-content: space-between;
+    .addYear {
+      width: 14rem;
+    }
+  }
+
+  .mainLayout {
```
Comment (anthopark) [inline new-nextjs-client/pages/index.module.scss]: .mainSection would be more appropriate

File: new-nextjs-client/pages/index.module.scss [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
   padding: 0 2rem;
   background: $main-background;
+  display: grid;
+  grid-template-rows: 7rem auto;
+
+  .topLayout {
+    display: flex;
+    justify-content: space-between;
+    .addYear {
+      width: 14rem;
```
Comment (anthopark) [inline new-nextjs-client/pages/index.module.scss]: The width should be defined in AddYearDropdown component.
Define this at the top level div container for AddYearDropdown. 
index.tsx should not have knowledge of the width of AddYearDropdown component.

**Review by anthopark — CHANGES_REQUESTED** [review]
Please revisit my comments

File: new-nextjs-client/src/components/IndexPage/AddYearDropdown/AddYearDropdown.tsx [hunk 5]
```diff
@@ -1,30 +1,39 @@
...
-    <div>
+    <div className={styles.container}>
       <AppSingleSelect
         isClearable={false}
         isOptionDisabled={(option: YearOption) => option.disabled}
         onChange={onSelectChange}
-        options={options}
-        placeholder="Add years..."
+        options={getYearOptions()}
+        placeholder="Add year"
```
Comment (anthopark) [inline new-nextjs-client/src/components/IndexPage/AddYearDropdown/AddYearDropdown.tsx]: `years`

**Review by anthopark — APPROVED** [review]

**Pull request merged**